card/mod/tinymce_editor/set/all/reference_editor/nest_image.rb

```diff
@@ -0,0 +1,75 @@
+format :html do
+  # view :nest_image, wrap: { modal: { footer: "" } }, unknown: true do
+  #   @nest_snippet = NestParser.new_image card.name.field("image01")
+  #   nest card.autoname(card.name.field("image01")), view: :new_image, type: :image
+  # end
+  #
+  view :nest_image, unknown: true, cache: :never,
+                    wrap: {
+                      slot: { class: "_overlay d0-card-overlay card nodblclick" }
+                    } do
+    nest_image_editor :overlay
+  end
+
+  view :modal_nest_image, unknown: true, cache: :never,
+                          wrap: { slot: { class: "nodblclick" } } do
+    nest_image_editor :modal
+  end
+
+  view :new_image, perms: :create, unknown: true, cache: :never do
+    new_view_frame_and_form new_image_form_opts
+  end
+
+  def nest_image_editor editor_mode
+    @tm_snippet_editor_mode = editor_mode
+    nest_name = card.autoname(card.name.field("image01"))
+    image_name = nest_name.to_name.right
+    @nest_snippet = NestEditor::NestParser.new_image image_name
+
+    voo.show! :content_tab
+    @nest_content_tab = nest(nest_name, view: :new_image, type: :image, hide: :guide)
```

Comment:
This is cognitively complex, but it's not obvious how to simplify it.  It looks like it was a tricky problem, so maybe a comment to explain why it was tricky would help.